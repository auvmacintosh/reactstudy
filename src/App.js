import React, {useState} from 'react';
import axios from 'axios';

const renderItem = article => (
    <h5 key={article._links.self.href}><a href={article._links.self.href}>{article.title}</a></h5>
);

const App = ({fuck}) => {
    const [articles, setArticles] = useState([]);

    const handleSearch = e => {
        e.preventDefault();
        axios.get('http://localhost:8080/api/articles').then(
            // res=>console.log(res)
            ({data: {_embedded: {articles}}}) => {
                setArticles(articles);
            }
        ).catch(
            (error) => {
                console.log(error)
            }
        );
    };

    return (
        <>
            {fuck}
            <button name='searchButton' onClick={handleSearch}>Search</button>
            <div data-testid="search-result">{articles.map(renderItem)}</div>
        </>
    );
};

export default App;