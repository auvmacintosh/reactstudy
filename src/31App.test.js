import React from 'react';
import ReactDOM from 'react-dom';
import {act} from 'react-dom/test-utils'
import App from './31App'

test('should tick to a new value', () => {
    jest.useFakeTimers();
    const el = document.createElement('div');
    // act(() => {
        ReactDOM.render(<App/>, el);
    // })
    expect(el.innerHTML).toBe("0");
    act(() => {
        jest.runAllTimers();
    })
    expect(el.innerHTML).toBe('1');
})

xtest('should increment counter', () => {
    const el = document.createElement('div');
    console.log(global === window)
    document.body.appendChild(el); // we attach the element to document.body to ensure events work
    ReactDOM.render(<App/>, el);
    const button = el.childNodes[0];
    act(() => {
        for (let i = 0; i < 3; i++) {
            button.dispatchEvent(new MouseEvent('click', {bubbles: true}));
        }
    });
    expect(button.innerHTML).toBe('3');
})