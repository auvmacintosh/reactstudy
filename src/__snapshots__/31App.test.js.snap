// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  "fuck",
  <button
    name="searchButton"
    onClick={[Function]}
  >
    Search
  </button>,
  <div
    data-testid="search-result"
  />,
]
`;
